x-pack/plugins/security_solution/server/lib/detection_engine/signals/source_fields_merging/utils/is_ignored.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/**
+ * Matches against anything you want to ignore and if it matches that field is ignored.
+ * @param fieldsKey The fields key to match against
+ * @param ignoreFields Array of fields to ignore. If a value starts and ends with "/", such as: "/[_]+/" then the field will be treated as a regular expression.
+ * If you have an object structure to ignore such as "{ a: { b: c: {} } } ", then you need to ignore it as the string "a.b.c"
+ * @returns true if it is a field to ignore, otherwise false
+ */
+export const isIgnored = (fieldsKey: string, ignoreFields: string[]): boolean => {
+  return ignoreFields.some((ignoreField) => {
+    if (ignoreField.startsWith('/') && ignoreField.endsWith('/')) {
+      return new RegExp(ignoreField.slice(1, -1)).test(fieldsKey);
```

Comment:
I think `new RegExp` can throw an error if the expression is invalid. Should we wrap this in a `try/catch` and maybe log that the expression was invalid if it fails? Or will it be bubbled up and logged elsewhere?

Reply:
I agree, and since it is user input data, I think I should validate the user input with a customer validator and custom message if 1 or more of the regular expressions do not work.

Just added a commit for this with tests around it:
https://github.com/elastic/kibana/pull/110927/commits/e755c3ed24d0ef5bf6648c7fa6cde8344d68d93e

If it's ok I don't want to do additional checks within here but rather trust the user input has been validated correctly and that Kibana is not allowed to start without proper validation which should be like the other code I see in the `config.ts` so we don't get users confused about why their regular expressions aren't working for them. Instead we will fail fast at startup if the regular expression is not valid.